File: src/test/fixtures/csharp/coherence/StaticBoundCallables.cs
using System;

class StaticOperations
{
    static Func<DateTime, DateTime> ParseDate = (DateTime value) => value;
    static Func<decimal, string> FormatNumber = (decimal value) => value.ToString();
    static Predicate<Guid> TestIdentifier = (Guid value) => value != Guid.Empty;
}

File: src/test/fixtures/cpp/coherence/static_bound_callables.cpp
#include <functional>
#include <string>

class StaticOperations {
public:
    inline static std::function<int(int)> parseNumber = [](int value) { return value; };
    inline static std::function<std::string(double)> formatNumber = [](double value) { return std::to_string(value); };
    inline static std::function<bool(char)> testCharacter = [](char value) { return value != 0; };
};
